include/deal.II/grid/manifold.h

```diff
@@ -773,6 +773,24 @@ class FlatManifold : public Manifold<dim, spacedim>
   get_tangent_vector (const Point<spacedim> &x1,
                       const Point<spacedim> &x2) const override;
 
+  /**
+   * Return the normal vector to the given face at point p taking into account
+   * that the face might not be flat.
```

Comment:
I think to avoid confusion, maybe say "that quadrilateral faces of hexahedral cells in 3d may not be planar". I don't think that any other case are affected, right? If so, maybe also add something like "In those cases, the face is assumed to have a geometry described by a bilinear function, and the normal vector is computed by embedding this bilinear form into a Cartesian space with a flat metric."

Same for the next function.